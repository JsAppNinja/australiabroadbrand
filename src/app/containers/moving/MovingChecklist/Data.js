export default {
  checkpoints: [
    {
      desc:
        "if you're renting,complete the property's condition report-it's a good idea to take photos and email then to your landloard or property manger",
    },
    {
      desc:
        'make sure all your utilities are connected(including your belong broadband) and the hot water service is on',
    },
    {
      desc:
        "if you're moving some of your possessions out of storage interstate or overseas,check your inventory-contact the moving company innediately if anything is missing",
    },
  ],
};
