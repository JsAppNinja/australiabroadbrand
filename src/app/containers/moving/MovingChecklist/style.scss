.checklist-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: #f6f9ff;
  padding: 110px 0;
  &__card {
    width: 1000px;
    box-sizing: border-box;
    position: relative;
    border-radius: 7px;
    padding: 40px 100px;
    background: #fff;
    &:not(:last-child) {
      margin-bottom: 60px;
    }
    &_orange {
      border: 1px solid #f89729;
    }
    &_green {
      border: 1px solid #4dbba6;
    }
    &_preview {
      font-family: Roboto;
      font-size: 14px;
      line-height: 24px;
      color: #313541;
    }
    &__content {
      .checkpoint {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        &__desc {
          margin-left: 15px;
          text-align: left;
          font-family: Roboto;
          font-size: 14px;
          line-height: 1.71;
          color: #313541;
        }
        .svg-inline--fa {
          color: #4dbba6;
        }
      }
    }
    &__footer {
      position: absolute;
      bottom: -15px;
      left: 0;
      right: 0;
      margin: auto;
      display: flex;
      flex-direction: column;
      align-items: center;
      cursor: pointer;
      span {
        margin-bottom: 5px;
        font-weight: bold;
      }
      .circle {
        width: 30px;
        height: 30px;
        border-radius: 50%;
        .svg-inline--fa {
          color: #fff;
        }
        &_orange {
          background: #f89729;
        }
        &_green {
          background: #67cebb;
        }
      }
    }
  }
  &__header {
    font-family: Baloo;
    font-size: 20px;
    letter-spacing: 1px;
    &_orange {
      color: #f89729;
    }
    &_green {
      color: #4dbba6;
    }
  }
}
