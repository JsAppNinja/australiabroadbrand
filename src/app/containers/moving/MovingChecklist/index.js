import MovingChecklist from './MovingChecklist';

export default MovingChecklist;
