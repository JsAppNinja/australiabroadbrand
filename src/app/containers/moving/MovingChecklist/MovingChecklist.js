import React, { Component } from 'react';
import { Arrow } from '../../../components/Arrow';

import HouseImg from '../../../assets/moving-checklist-house.png';
import ChecklistImg from '../../../assets/moving-checklist.png';
import './style.scss';

import Data from './Data';

class MovingChecklist extends Component {
  state = {
    isOrangeExpanded: false,
    isGreenExpanded: false,
  };
  expandCard = type => {
    const value = `is${type}Expanded`;
    this.setState({
      [value]: !this.state[value],
    });
  };
  render() {
    const { isGreenExpanded, isOrangeExpanded } = this.state;
    return (
      <section className="checklist-section">
        <article className="checklist-section__card checklist-section__card_orange">
          <img src={HouseImg} alt="" />
          <h2 className="checklist-section__header checklist-section__header_orange">
            Moving from your old home
          </h2>
          <p className="checklist-section__card_preview">
            Most people agree - moving house is a real headache! To make things
            easier, we've put together a mover's checklist so you can tick your
            moving jobs off as you go.
          </p>
          {isOrangeExpanded ? (
            <div className="checklist-section__card__content">data</div>
          ) : null}
          <div
            className="checklist-section__card__footer"
            onClick={() => this.expandCard('Orange')}
          >
            <span>{isOrangeExpanded ? 'Show less' : 'Show more'}</span>
            <div className="circle circle_orange">
              <Arrow rotate={isOrangeExpanded ? true : false} />
            </div>
          </div>
        </article>
        <article className="checklist-section__card checklist-section__card_green">
          <img src={ChecklistImg} alt="" />
          <h2 className="checklist-section__header checklist-section__header_green">
            At your new home
          </h2>
          <p className="checklist-section__card_preview">
            Lorem ipsum, dolor sit amet consectetur adipisicing elit.
            Temporibus, explicabo!
          </p>
          {isGreenExpanded ? (
            <div className="checklist-section__card__content">
              {Data.checkpoints.map((item, index) => (
                <div className="checkpoint" key={index}>
                  <Arrow />
                  <p className="checkpoint__desc">{item.desc}</p>
                </div>
              ))}
            </div>
          ) : null}
          <div
            className="checklist-section__card__footer"
            onClick={() => this.expandCard('Green')}
          >
            <span>{isGreenExpanded ? 'Show less' : 'Show more'}</span>
            <div className="circle circle_green">
              <Arrow rotate={isGreenExpanded ? true : false} />
            </div>
          </div>
        </article>
      </section>
    );
  }
}

export default MovingChecklist;
