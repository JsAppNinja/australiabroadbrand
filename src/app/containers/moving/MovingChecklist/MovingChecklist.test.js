import React from 'react';
import { shallow, mount } from 'enzyme';

import MovingChecklist from './MovingChecklist.js';

describe('MovingChecklist tests', () => {
  const tree = shallow(<MovingChecklist />);
  const spyOnToggleSteps = jest.spyOn(tree.instance(), 'expandCard');
  it('should render moving checklist section', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should handle click on checklist', () => {
    const btnOrange = tree.find(
      '.checklist-section__card_orange .checklist-section__card__footer'
    );
    const btnGreen = tree.find(
      '.checklist-section__card_green .checklist-section__card__footer'
    );

    btnOrange.simulate('click', 'Orange');
    expect(spyOnToggleSteps).toHaveBeenCalled();
    expect(tree.state().isOrangeExpanded).toEqual(true);
    btnGreen.simulate('click', 'Green');
    expect(spyOnToggleSteps).toHaveBeenCalled();
    expect(tree.state().isGreenExpanded).toEqual(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<MovingChecklist />).find('.checklist-section').length).toBe(
      1
    );
  });
});
