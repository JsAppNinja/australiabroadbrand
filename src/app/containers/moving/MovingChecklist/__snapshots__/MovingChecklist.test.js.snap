// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovingChecklist tests should render moving checklist section 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovingChecklist />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="checklist-section__card checklist-section__card_orange"
        >
          <img
            alt=""
            src="moving-checklist-house.png"
          />
          <h2
            className="checklist-section__header checklist-section__header_orange"
          >
            Moving from your old home
          </h2>
          <p
            className="checklist-section__card_preview"
          >
            Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.
          </p>
          <div
            className="checklist-section__card__footer"
            onClick={[Function]}
          >
            <span>
              Show more
            </span>
            <div
              className="circle circle_orange"
            >
              <Arrow
                rotate={false}
              />
            </div>
          </div>
        </article>,
        <article
          className="checklist-section__card checklist-section__card_green"
        >
          <img
            alt=""
            src="moving-checklist.png"
          />
          <h2
            className="checklist-section__header checklist-section__header_green"
          >
            At your new home
          </h2>
          <p
            className="checklist-section__card_preview"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!
          </p>
          <div
            className="checklist-section__card__footer"
            onClick={[Function]}
          >
            <span>
              Show more
            </span>
            <div
              className="circle circle_green"
            >
              <Arrow
                rotate={false}
              />
            </div>
          </div>
        </article>,
      ],
      "className": "checklist-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              src="moving-checklist-house.png"
            />,
            <h2
              className="checklist-section__header checklist-section__header_orange"
            >
              Moving from your old home
            </h2>,
            <p
              className="checklist-section__card_preview"
            >
              Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.
            </p>,
            null,
            <div
              className="checklist-section__card__footer"
              onClick={[Function]}
            >
              <span>
                Show more
              </span>
              <div
                className="circle circle_orange"
              >
                <Arrow
                  rotate={false}
                />
              </div>
            </div>,
          ],
          "className": "checklist-section__card checklist-section__card_orange",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "moving-checklist-house.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Moving from your old home",
              "className": "checklist-section__header checklist-section__header_orange",
            },
            "ref": null,
            "rendered": "Moving from your old home",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.",
              "className": "checklist-section__card_preview",
            },
            "ref": null,
            "rendered": "Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.",
            "type": "p",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Show more
                </span>,
                <div
                  className="circle circle_orange"
                >
                  <Arrow
                    rotate={false}
                  />
                </div>,
              ],
              "className": "checklist-section__card__footer",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Show more",
                },
                "ref": null,
                "rendered": "Show more",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Arrow
                    rotate={false}
                  />,
                  "className": "circle circle_orange",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "rotate": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              src="moving-checklist.png"
            />,
            <h2
              className="checklist-section__header checklist-section__header_green"
            >
              At your new home
            </h2>,
            <p
              className="checklist-section__card_preview"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!
            </p>,
            null,
            <div
              className="checklist-section__card__footer"
              onClick={[Function]}
            >
              <span>
                Show more
              </span>
              <div
                className="circle circle_green"
              >
                <Arrow
                  rotate={false}
                />
              </div>
            </div>,
          ],
          "className": "checklist-section__card checklist-section__card_green",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "moving-checklist.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "At your new home",
              "className": "checklist-section__header checklist-section__header_green",
            },
            "ref": null,
            "rendered": "At your new home",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!",
              "className": "checklist-section__card_preview",
            },
            "ref": null,
            "rendered": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!",
            "type": "p",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Show more
                </span>,
                <div
                  className="circle circle_green"
                >
                  <Arrow
                    rotate={false}
                  />
                </div>,
              ],
              "className": "checklist-section__card__footer",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Show more",
                },
                "ref": null,
                "rendered": "Show more",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Arrow
                    rotate={false}
                  />,
                  "className": "circle circle_green",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "rotate": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="checklist-section__card checklist-section__card_orange"
          >
            <img
              alt=""
              src="moving-checklist-house.png"
            />
            <h2
              className="checklist-section__header checklist-section__header_orange"
            >
              Moving from your old home
            </h2>
            <p
              className="checklist-section__card_preview"
            >
              Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.
            </p>
            <div
              className="checklist-section__card__footer"
              onClick={[Function]}
            >
              <span>
                Show more
              </span>
              <div
                className="circle circle_orange"
              >
                <Arrow
                  rotate={false}
                />
              </div>
            </div>
          </article>,
          <article
            className="checklist-section__card checklist-section__card_green"
          >
            <img
              alt=""
              src="moving-checklist.png"
            />
            <h2
              className="checklist-section__header checklist-section__header_green"
            >
              At your new home
            </h2>
            <p
              className="checklist-section__card_preview"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!
            </p>
            <div
              className="checklist-section__card__footer"
              onClick={[Function]}
            >
              <span>
                Show more
              </span>
              <div
                className="circle circle_green"
              >
                <Arrow
                  rotate={false}
                />
              </div>
            </div>
          </article>,
        ],
        "className": "checklist-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                src="moving-checklist-house.png"
              />,
              <h2
                className="checklist-section__header checklist-section__header_orange"
              >
                Moving from your old home
              </h2>,
              <p
                className="checklist-section__card_preview"
              >
                Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.
              </p>,
              null,
              <div
                className="checklist-section__card__footer"
                onClick={[Function]}
              >
                <span>
                  Show more
                </span>
                <div
                  className="circle circle_orange"
                >
                  <Arrow
                    rotate={false}
                  />
                </div>
              </div>,
            ],
            "className": "checklist-section__card checklist-section__card_orange",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "moving-checklist-house.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Moving from your old home",
                "className": "checklist-section__header checklist-section__header_orange",
              },
              "ref": null,
              "rendered": "Moving from your old home",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.",
                "className": "checklist-section__card_preview",
              },
              "ref": null,
              "rendered": "Most people agree - moving house is a real headache! To make things easier, we've put together a mover's checklist so you can tick your moving jobs off as you go.",
              "type": "p",
            },
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Show more
                  </span>,
                  <div
                    className="circle circle_orange"
                  >
                    <Arrow
                      rotate={false}
                    />
                  </div>,
                ],
                "className": "checklist-section__card__footer",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Show more",
                  },
                  "ref": null,
                  "rendered": "Show more",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Arrow
                      rotate={false}
                    />,
                    "className": "circle circle_orange",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "rotate": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                src="moving-checklist.png"
              />,
              <h2
                className="checklist-section__header checklist-section__header_green"
              >
                At your new home
              </h2>,
              <p
                className="checklist-section__card_preview"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!
              </p>,
              null,
              <div
                className="checklist-section__card__footer"
                onClick={[Function]}
              >
                <span>
                  Show more
                </span>
                <div
                  className="circle circle_green"
                >
                  <Arrow
                    rotate={false}
                  />
                </div>
              </div>,
            ],
            "className": "checklist-section__card checklist-section__card_green",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "moving-checklist.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "At your new home",
                "className": "checklist-section__header checklist-section__header_green",
              },
              "ref": null,
              "rendered": "At your new home",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!",
                "className": "checklist-section__card_preview",
              },
              "ref": null,
              "rendered": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, explicabo!",
              "type": "p",
            },
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Show more
                  </span>,
                  <div
                    className="circle circle_green"
                  >
                    <Arrow
                      rotate={false}
                    />
                  </div>,
                ],
                "className": "checklist-section__card__footer",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Show more",
                  },
                  "ref": null,
                  "rendered": "Show more",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Arrow
                      rotate={false}
                    />,
                    "className": "circle circle_green",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "rotate": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
